spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -340,14 +355,22 @@ public boolean enableBatchRead() {
 
       boolean onlyPrimitives = lazySchema().columns().stream().allMatch(c -> c.type().isPrimitiveType());
 
+      boolean hasTimestampWithoutZone = hasTimestampWithoutZone(lazySchema());
+
       boolean hasNoDeleteFiles = tasks().stream().noneMatch(TableScanUtil::hasDeletes);
 
       this.readUsingBatch = batchReadsEnabled && hasNoDeleteFiles && ((allOrcFileScanTasks && hasNoRowFilters) ||
-          (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives));
+          (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives && !hasTimestampWithoutZone));
```

Comment:
Could you please explain why this is required now (especially no special treatment existed for schemas with Timestamp before this patch)?

Reply:
I don't quite remember this because it was a while ago, but my guess is that vectorized reads in Parquet do not support timestamp without zone type. Parquet vectorized reads are only supported in Spark and so my guess is that the implementation did not bother supporting it.